spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,438 @@
+
+    transformer_name = _resolve_curated_trf_pipe_name(msg, config, cli_transformer_name)
+    model_src = _resolve_model_source(msg, cli_model_name, cli_model_revision)
+    model_name, model_revision = _resolve_model_name_and_revision(
+        msg, config, model_src, cli_model_name, cli_model_revision, transformer_name
+    )
+
+    hf_model_type = _lookup_hf_model_type_for_curated_architecture(
+        msg, config, transformer_name
+    )
+    hf_config = _get_hf_model_config(model_name, model_revision)
+    incoming_hf_model_type = hf_config["model_type"]
+    if incoming_hf_model_type != hf_model_type:
+        msg.fail(
+            f"Hugging Face model of type '{incoming_hf_model_type}' cannot be loaded into "
+            f"Curated Transformer pipe '{transformer_name}' due to mismatching architectures",
+            exits=1,
+        )
+
+    params_to_fill = COMMON_ENTRYPOINT_PARAMS.copy()
+    params_to_fill.update(MODEL_SPECIFIC_ENTRYPOINT_PARAMS.get(hf_model_type, {}))
+    hf_tokenizer = _load_hf_tokenizer(model_name, model_revision, msg)
+
+    filled_params = _fill_parameters(msg, params_to_fill, hf_config, hf_tokenizer)
+
+    # Update transformer model config with the filled params.
+    trf_config = config["components"][transformer_name]["model"]
+    trf_config.update(filled_params)
+
+    if model_src == ModelSource.CliArgument:
+        # Overwrite the encoder and piece loader configs.
+        _save_encoder_loader_config(
+            msg, config, transformer_name, model_name, model_revision
+        )
+        _save_piece_loader_config(
+            msg, config, transformer_name, model_name, model_revision, overwrite=True
+        )
+    else:
+        # Only fill in the piece loader config if it's not present.
+        _save_piece_loader_config(
+            msg, config, transformer_name, model_name, model_revision, overwrite=False
+        )
+
+    is_stdout = str(output_path) == "-"
+    if is_stdout:
+        msg.info(title="\n\nOutput:")
+    save_config(config, output_path, is_stdout=is_stdout, silent=is_stdout)
+
+
+def _validate_hf_packages(msg: Printer):
+    if not has_huggingface_hub or not has_hf_transformers:
+        msg.fail(
+            "This command requires the `huggingface-hub` and `transformers` packages to be installed",
+            exits=1,
+        )
+
+
+def _resolve_model_source(
+    msg: Printer,
+    cli_model_name: Optional[str],
+    cli_model_revision: Optional[str],
+) -> ModelSource:
+    if cli_model_name is None and cli_model_revision is None:
+        return ModelSource.Loader
+
+    if cli_model_name is not None and cli_model_revision is not None:
+        return ModelSource.CliArgument
+
+    if (cli_model_name is None and cli_model_revision is not None) or (
+        cli_model_name is not None and cli_model_revision is None
+    ):
+        msg.fail(
+            "Either both model name and model revision are provided as command-line arguments or none of ther are",
```

Comment:
Probably to be modified, but:
```suggestion
            "Either both model name and model revision are provided as command-line arguments or neither of them are",
```